src/lib/parse-cdn.ts

```diff
@@ -1,31 +1,51 @@
 import { embeddedWriteKey } from './embedded-write-key'
-const regex = /(https:\/\/.*)\/analytics\.js\/v1\/(?:.*?)\/(?:platform|analytics.*)?/
-
-export function getCDN(): string {
-  let cdn: string | undefined = undefined
 
+const analyticsScriptRegex = /(https:\/\/.*)\/analytics\.js\/v1\/(?:.*?)\/(?:platform|analytics.*)?/
+const getCDNUrlFromScriptTag = (): string | undefined => {
+  let cdn: string | undefined
   const scripts = Array.prototype.slice.call(
     document.querySelectorAll('script')
   )
-
   scripts.forEach((s) => {
     const src = s.getAttribute('src') ?? ''
-    const result = regex.exec(src)
+    const result = analyticsScriptRegex.exec(src)
 
     if (result && result[1]) {
       cdn = result[1]
     }
   })
+  return cdn
+}
 
-  // it's possible that the CDN is not found in the page because:
-  // - the script is loaded through a proxy
-  // - the script is removed after execution
-  // in this case, we fall back to the default Segment CDN
-  if (!cdn) {
-    return `https://cdn.segment.com`
+const getGlobalCDNUrl = (): string | undefined => {
+  if (window.analytics) {
+    return window.analytics._cdn
   }
+}
 
-  return cdn
+export const setGlobalCDNUrl = (cdn: string) => {
+  if (window.analytics) {
+    window.analytics._cdn = cdn
+  }
+}
+
+export const getCDN = (): string => {
+  const globalCdnUrl = getGlobalCDNUrl()
+
+  if (globalCdnUrl) return globalCdnUrl
+
+  const cdnFromScriptTag = getCDNUrlFromScriptTag()
+
+  if (cdnFromScriptTag) {
+    setGlobalCDNUrl(cdnFromScriptTag)
```

Comment:
Personally, keeping CQS in mind, I rather move this to `setGlobalCDNUrl` where the `cdn` could be `undefined` or `null`.

```ts
export const setGlobalCDNUrl = (cdn?: string) => {
  if (window.analytics) {
    window.analytics._cdn = cdn ?? getCDNUrlFromScriptTag()
  }
}
```

You deal with the command side once when you are trying to set things up.

Especially that, if I am not mistaken, ideally `getCDNUrlFromScriptTag` would be removed altogether.

For example,

```ts
import { getCDNUrlFromScriptTag } from '...';
AnalyticsBrowser.load({
      cdn: getCDNUrlFromScriptTag(), // notice that I am using when I initialize things
      writeKey
})
```

Or at least I hope Segment move to that, so the bundle sizes are taken into consideration.

Reply:
> I’ve gone back to explicitly setting the global


I love it!

Reply:
> @silesky one of the reasons why I didn't finish it right away the way you did it was because I wanted to figure out how to remove that `window.analytics._cdn`.
> 
> I am curious about the technical limitations but would be much better to avoid having that global object if possible. I am not sure.
> 
> Any thoughts on that?

I agree that the global is ugly, and I was able to further minimize its footprint by removing  the setGlobal call from the load-browser call (after doing some digging, looking at tests, and being satisfied that there weren't any more implicit dependencies.)

The window.analytics._cdn is the current source of truth for standalone.ts.  and, while maybe undocumented, is part of the existing API (i.e. being able to set window.analytics._writeKey the same way). We couldn't get rid of it without breaking downstream consumers, so it's probably here to stay.